File: SplitArray.rb
n, m = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)
b = gets.split.map(&:to_i)

nextIndex = 0
b.each do |x|
    x.times do |i|
        print a[nextIndex]
        if i < x - 1
            print " "
        else
            puts
        end

        nextIndex += 1
    end
end

File: SplitArray.py
n, m = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

nextIndex = 0
for x in b:
    for i in range(x):
        print(a[nextIndex], end="")
        if i < x - 1:
            print(" ", end="")
        else:
            print()
        nextIndex += 1

